### PR TITLE
Adding release workflow

### PR DESCRIPTION
**Pull request by leahwicz**
### Description

This is a GitHub Action that tests, builds, and packages `dbt-core` and `dbt-postgres` and then releases them to GitHub and PyPi.

Example workflow: https://github.com/leahwicz/dbt/actions/runs/1468971423
Example release: https://github.com/leahwicz/dbt/releases

I would like to try this out for RC2 🎉 

### Checklist

- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    shell: bash
+
+jobs:
+  unit:
+    name: Unit test
+
+    runs-on: ubuntu-latest
+
+    env:
+      TOXENV: "unit"
```
Comment (leahwicz) [inline .github/workflows/release.yml:34]: Is unit testing enough verification for moving forward with the release or should we run some integration test? We should only be releasing when the last commit is all 🟢 in Actions but that relies on us doing our due diligence and checking

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+          echo ::set-output name=isPrerelease::$IS_PRERELEASE
+
+      - name: Creating GitHub Release
+        uses: softprops/action-gh-release@v1
+        with:
+          name: dbt-core v${{github.event.inputs.version_number}}
+          tag_name: v${{github.event.inputs.version_number}}
+          prerelease: ${{ steps.release_type.outputs.isPrerelease }}
+          target_commitish: ${{github.event.inputs.sha}}
+          body: |
```
Comment (leahwicz) [inline .github/workflows/release.yml:173]: We will still have to paste in the release notes we want manually but for now that seemed like an ok start. We have been talking about changing how we do the Changelog anyway so maybe once we have a plan there we can figure how to automatically pull them in

File: .github/workflows/release.yml (hunk 1 above)
Comment (jtcohen6) [reply .github/workflows/release.yml:34]: Unit tests should catch the very obvious issues. Of course, if there are failing integration tests on the `main`/release branch, we should know about them

File: .github/workflows/release.yml (hunk 2 above)
Comment (jtcohen6) [reply .github/workflows/release.yml:173]: That's fine. We can also edit the GitHub release, and update release notes after the fact

**Pull request merged**